lib/saluki-components/src/sources/checks/scheduler.rs

```diff
@@ -0,0 +1,506 @@
+        let worker_channels = Arc::clone(&self.worker_channels);
+
+        // Create and spawn the ticker task
+        let handle = tokio::spawn(async move {
+            let mut ticker = time::interval(interval_duration);
+            ticker.tick().await; // Skip the first immediate tick
+
+            loop {
+                // Wait for the next interval tick
+                ticker.tick().await;
+
+                // Get all checks in this interval
+                let check_ids = {
+                    let buckets = interval_buckets.read().unwrap();
+                    match buckets.get(&interval_secs) {
+                        Some(bucket) => bucket.iter().cloned().collect::<Vec<_>>(),
+                        None => {
+                            // This bucket no longer exists, exit the ticker
+                            debug!("Interval {}s no longer has any checks, stopping ticker", interval_secs);
+                            break;
+                        }
+                    }
+                };
+
+                let channels = {
+                    let channels_guard = worker_channels.lock().unwrap();
+                    if channels_guard.is_empty() {
+                        continue; // No workers available
+                    }
+                    channels_guard.clone()
+                };
+
+                // Queue each check for execution using round-robin distribution
+                let channel_count = channels.len();
+                for (i, check_id) in check_ids.into_iter().enumerate() {
+                    let check = {
+                        let checks_map = checks.read().unwrap();
+                        match checks_map.get(&check_id) {
+                            Some(check) => Arc::clone(check),
+                            None => continue,
+                        }
+                    };
+
+                    // Simple round-robin: select channel based on check index
+                    let channel_idx = i % channel_count;
+                    let channel = &channels[channel_idx];
+
+                    // Send to worker
+                    if let Err(e) = channel.try_send(WorkerMessage::RunCheck(check)) {
+                        error!("Failed to queue check '{}': {}", check_id, e);
+                    }
+                }
+            }
+        });
+
+        let mut tickers = self.interval_handles.lock().unwrap();
+        tickers.insert(interval_secs, handle);
+
+        debug!("Started ticker for interval {}s", interval_secs);
+    }
+
+    /// Stop the ticker for a specific interval
+    fn stop_interval_ticker(&self, interval_secs: u64) {
+        let mut tickers = self.interval_handles.lock().unwrap();
+        if let Some(handle) = tickers.remove(&interval_secs) {
+            handle.abort();
+            debug!("Stopped ticker for interval {}s", interval_secs);
+        }
+    }
+
+    /// Adjust the number of workers
+    fn adjust_worker_count(&self, desired_count: usize) {
```

Comment:
I'm a bit skeptical that "number of workers" is a useful construct for the python check runners.

Since python execution is blocked on the GIL anyway, there is no "concurrency" here.
And even if we look ahead to a `nogil` future, its not clear to me why we would manage our own runner abstraction level when we can defer this to `tokio` for scheduling.

Reply:
@scottopell that is a very valid point. 

Still, I want to understand this last part for your comment:
> And even if we look ahead to a nogil future, its not clear to me why we would manage our own runner abstraction level when we can defer this to tokio for scheduling.

Each scheduler worker is a tokio thread, that which would be manage by the tokio scheduler https://github.com/DataDog/saluki/blob/883e35684c91f2361fbf78cbdbb5770d87af2e64/lib/saluki-components/src/sources/checks/scheduler.rs#L254

Are you suggesting to not even worry about having workers at all and running each check on a tokio thread, and let tokio handle everything?